styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java

```diff
@@ -97,10 +99,11 @@ void tick() {
 
       if (enabled.contains(workflow.id())) {
         try {
-          triggerListener.event(
+          CompletionStage<Void> processed = triggerListener.event(
               workflow,
               Trigger.natural(),
               decrementInstant(next, partitioning));
+          processed.toCompletableFuture().get(1, TimeUnit.MINUTES);
```

Comment:
Waiting forever should be equivalent to executing the event handler and resulting state transitions etc synchronously. I believe @rouzwawi agreed that this is what we want to do but let's verify =)